src/uwtools/config/tools.py

```diff
@@ -112,8 +112,8 @@ def _realize_config_check_depths(input_obj: Config, output_format: str) -> None:
     :param output_format: The output format:
     :raises: UWConfigError on excessive input-config depth.
     """
-    if (output_format == FORMAT.ini and input_obj.depth > 2) or (
-        output_format == FORMAT.nml and input_obj.depth != 2
+    if (output_format == (FORMAT.ini or FORMAT.nml) and input_obj.depth != 2) or (
+        output_format == FORMAT.sh and input_obj.depth != 1
```

Comment:
Seeing this, I'm wondering if the `_MAXDEPTH` attribute of the format classes should be `MAXDEPTH` instead, and those values used here.

I'm also wondering if the `MAX` part is what we really want, as it's also `MIN`, right? Just thinking out loud but, except for YAML, which AFAIK has no maximum depth, the other formats all have a specific depth that they must be, neither more or less. Should we just call it `DEPTH`?

Reply:
Even more annoyingly, this goes directly to FORMAT without accessing the concrete class, which is why I ended going back to leaving it specific and not just `input_obj._MAXDEPTH`. I actually added a limit to YAML to cheekily stop absurd subtask-ing, but that could be dropped and just leave it as _DEPTH

I thought about how to call it here, but I think it would have to be modified to accept the concrete class and do the evaluation, or move DEPTH to a property in FORMAT

Reply:
Not sure if we're on the same page, but I was was thinking of something like
```
if (output_format == FORMAT.ini and input_obj.depth != INIConfig.DEPTH) or
    (output_format == FORMAT.nml and input_obj.depth != NMLConfig.DEPTH) or
    (output_format == FORMAT.sh and input_obj.depth != SHConfig.DEPTH): ...
```
But we've also got `uwtools.config.support.format_to_config()` at our disposal, so I think we could have something like
```
cfgclass = format_to_config(output_format)
if cfgclass.DEPTH and input_obj.depth != cfgclass.DEPTH: ...
```
The first clause in the `if` accounts for the fact that `YAMLConfig` should have `DEPTH` of `None`, and the second clause catches depth mismatch. Would that work?

Reply:
Ah, yes - I had at first tried to go with `if(input_obj.depth !<= input_obj._MAXDEPTH)` for a clean single line, but that was the wrong object type. I think using `format_to_config` is the right way, though I think we can leave it to just `if input_obj.depth != cfgclass.DEPTH:`

Reply:
Nope, you're right, got to define None depths